test/users-create-update.test.js

```diff
@@ -128,56 +129,199 @@ describe('POST /users/:safeAddress - Updating user data', () => {
+        [payload.address, payload.data.safeAddress, similarUserName],
+        privateKey,
+      );
+      // Update payload values
+      payload.data.username = similarUserName;
+      payload.signature = signature;
+
+      mockGraphUsers(payload.address, payload.data.safeAddress);
+      await request(app)
+        .post(`/api/users/${payload.data.safeAddress}`)
+        .send({
+          address: payload.address,
+          signature: payload.signature,
+          data: payload.data,
+        })
+        .set('Accept', 'application/json')
+        .expect(httpStatus.OK);
+    });
+
+    it('should successfully respond when I update only the avatarUrl', async () => {
+      await request(app)
+        .put('/api/users')
+        .send(payload)
+        .set('Accept', 'application/json')
+        .expect(httpStatus.CREATED);
+
+      const signature = getSignature(
+        [payload.address, payload.data.safeAddress, payload.data.username],
+        privateKey,
+      );
+      // Update payload values
+      payload.data.avatarUrl = newAvatarUrl;
+      payload.signature = signature;
+
+      mockGraphUsers(payload.address, payload.data.safeAddress);
+      await request(app)
+        .post(`/api/users/${payload.data.safeAddress}`)
+        .send({
+          address: payload.address,
+          signature: payload.signature,
+          data: payload.data,
+        })
+        .set('Accept', 'application/json')
+        .expect(httpStatus.OK);
+    });
+
+    it('should successfully respond when mail is missing', async () => {
+      await request(app)
+        .put('/api/users')
+        .send(payload)
+        .set('Accept', 'application/json')
+        .expect(httpStatus.CREATED);
+
+      const signature = getSignature(
+        [payload.address, payload.data.safeAddress, newUsername],
+        privateKey,
+      );
+      // Update payload values
+      payload.data.username = newUsername;
+      payload.data.avatarUrl = newAvatarUrl;
+      payload.signature = signature;
+
+      mockGraphUsers(payload.address, payload.data.safeAddress);
+      await request(app)
+        .post(`/api/users/${payload.data.safeAddress}`)
+        .send({
+          address: payload.address,
+          signature: payload.signature,
+          data: {
+            safeAddress: payload.data.safeAddress,
+            username: payload.data.username,
+            avatarUrl: payload.data.avatarUrl,
+          },
+        })
+        .set('Accept', 'application/json')
+        .expect(httpStatus.OK);
+    });
   });
 
-  it('should successfully respond when user was not registered', async () => {
-    const signature = getSignature(
-      [payload.address, payload.data.safeAddress, newUsername],
-      privateKey,
-    );
-    // Update payload values
-    payload.data.username = newUsername;
-    payload.data.email = newEmail;
-    payload.data.avatarUrl = newAvatarUrl;
-    payload.signature = signature;
+  describe('when user was not registered', () => {
```

Comment:
in this case, doesnt it mean that username and avatar are not saved on the first place so it should fail completely not only if mail is missing?

Reply:
In which case are you talking about?
When you say "in the first place" do you men in the registration?

Reply:
`describe('when user was not registered', () => {` yes, when a user tries to register

Reply:
Sometimes it can happen that you have a fully functional circles account but you don't have a username or avatarURL (i.e. you haven't registered in the users db). 
In those cases the api would let this users create an entry al long as the introduce an email (since the email is unique and not null in the db)

Reply:
I didn't want to limit the api to only update. because the upsert woul be useful for those users whose circles accounts are working but they don't have a register on the user profile db.

Reply:
Fixed now!